httpx/_client.py

```diff
@@ -1015,6 +1036,7 @@ def delete(
             timeout=timeout,
         )
 
+    @check_not_closed
```

Comment:
Could we instead just let things pass silently if the client is already closed, and `.close()` is called again?
So...

```python
if not self._is_closed:
    self._transport.close()
    for proxy in self._proxies.values():
        if proxy is not None:
            proxy.close()